modules/bishop.py

```diff
@@ -0,0 +1,55 @@
+import copy
+
+from modules.chess_piece import ChessPiece
+
+class Bishop(ChessPiece):
+    def __init__(self, position, color, id):
+        super().__init__(position, "bishop", color, id)
+        self.state["has_moved"] = False
+
+    def __deepcopy__(self, memodict={}):
+        deepcopy = Bishop(
+            copy.deepcopy(self.position),
+            copy.deepcopy(self.color),
+            copy.deepcopy(self.id),
+        )
+        deepcopy.state["has_moved"] = copy.deepcopy(self.state["has_moved"])
+
+        return deepcopy
+
+    def get_allowed_moves(self, board):  # NOT TESTABLE
+        super().get_allowed_moves(board)
+        x, y = self.position
+        allowed_moves = []
+
+        for dx in [-1, 1]:
+            for dy in [-1, 1]:
+                for dist in range(1,8):
```

Comment:
```suggestion
                for dist in range(1, max(board.board_shape)):
```
Could keep it more versatile by using the board_shape